format/puffin-spec.md

```diff
@@ -0,0 +1,144 @@
+---
+url: puffin-spec
```

Comment:
I merged https://github.com/apache/iceberg/pull/4944 so I think we can remove this.

Reply:
thanks for the merge! rebased